tests/test_diagnostics.py

```diff
@@ -2,6 +2,7 @@
 Tests for the diagonstics.py file
 """
 import pytest
+import shutil

```

Comment:
I'd like to not have to use `shutil` for these tests either. I think I need to look at this code more, though, before I can give specific recommendations.

A couple quick thoughts, though: I think there may be too many `assert` statements in `test_initialize_should_set_remaining_parameters_when_called`. It should probably be broken up. Also, the fixture seems a bit strange... why is the diagnostic not in the simulation configuration dictionary?